web-api/src/persistence/elasticsearch/helpers/searchQueryForAggregation.ts

```diff
@@ -0,0 +1,67 @@
+import { FetchEventCodesParamsType } from '../fetchEventCodesCountForJudges';
+import { QueryDslQueryContainer } from '@opensearch-project/opensearch/api/types';
+
+export const computeShouldFilters = ({
+  params,
+}: {
+  params: FetchEventCodesParamsType;
+}) => {
+  const shouldFilters: QueryDslQueryContainer[] = [];
+
+  if (params.judges) {
+    params.judges.forEach(judgeName => {
+      shouldFilters.push({
+        match: {
+          ['signedJudgeName.S']: {
+            operator: 'and',
+            query: judgeName,
+          },
+        },
+      });
+
+      shouldFilters.push({
+        match: {
+          ['judge.S']: judgeName,
+        },
+      });
+    });
+  }
+
+  return shouldFilters;
+};
+
+export const computeDocumentFilters = ({ params }) => {
+  // add filters to return 'served'/'stricken' orders/opinions
+  const documentFilters: QueryDslQueryContainer[] = [
+    { term: { 'entityName.S': 'DocketEntry' } },
+    {
+      exists: {
+        field: 'servedAt',
+      },
+    },
+    {
+      term: {
+        'isStricken.BOOL': false,
+      },
+    },
+  ];
+
+  if (params.endDate && params.startDate) {
+    documentFilters.push({
+      range: {
+        'filingDate.S': {
+          gte: `${params.startDate}||/h`,
+          lte: `${params.endDate}||/h`,
+        },
+      },
+    });
+  }
+
+  if (params.searchType) {
+    documentFilters.push({
+      terms: { 'eventCode.S': params.documentEventCodes },
+    });
+  }
```

Comment:
Is this the only way that `params.searchType` is used? Can it be removed? I see either `opinion` or `order` getting passed in to the function that uses this. Can't we simply rely on `params.documentEventCodes`?

Reply:
Resolved in this commit: [246dcc0](https://github.com/ustaxcourt/ef-cms/pull/3799/commits/246dcc03369049c305da6dcf8777a5e2b1ecc492)